### PR TITLE
Fix: Transferability with batch transfer

### PR DESCRIPTION
**Pull request by fabianschu**
- fixes bug
- introduces batch tier creation

### CODE REVIEW

File: contracts/governance/Badger.sol [hunk 1]
```diff
@@ -229,14 +253,30 @@ contract Badger is Ownable, ERC1155 {
         uint256 tokenId,
         string calldata uriId,
         bool transferable
-    ) public onlyOwner isValidString(uriId) {
+    ) public onlyOwner {
+        _createTokenTier(tokenId, uriId, transferable);
```
Comment (BritikovKI) [inline contracts/governance/Badger.sol:257]: Can't we just call this function inside of batchCreateTokenTiers instead of implementing internal function?

Comment (fabianschu) [reply contracts/governance/Badger.sol:257]: Yes we could, but then we would check `onlyOwner` in every iteration of the loop. I wanted to save on that. Does that make sense?

Comment (BritikovKI) [reply contracts/governance/Badger.sol:257]: Actually yes, it makes sense

**Review by BritikovKI — APPROVED** [review]
Lgtmt

**Comment by codebuster22** [comment]
Hey, rather than individually overriding each transfer function, I think we should just override the hook `_beforeTokenTransfer` that is called everytime before a token transfer happens (this includes all kind of transfers, including mint and burn)

Ref:- https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/ERC1155.sol#L405

**Review by Zitzak — APPROVED** [review]
lgtm

**Review by Zitzak — DISMISSED** [review]
LGTM

**Review by Zitzak — APPROVED** [review]
LGTM

**Review by marvinkruse — APPROVED** [review]
Good catch on this, @fabianschu - PR looks good

**Review by codebuster22 — APPROVED** [review]
lgtm

**Pull request merged**